frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1883,7 +1887,7 @@ function ReaderFooter:setTocMarkers(reset)
                 self.progress_bar.ticks = self.ui.toc:getTocTicksFlattened()
             end
             if self.view.view_mode == "page" then
-                self.progress_bar.last = self.pages or self.ui.document:getPageCount()
+                self.progress_bar.last = self.ui.doc_settings:readSetting("doc_pages_max") or self.ui.document:getPageCount()
```

Comment:
You removed `self.pages` - but by experience, this footer code is twisted, and it may very well be that it was needed - even if it _looks_ like `self.pages` is equal to `self.ui.document:getPageCount()`, it's possible that at edges (on load, on exit, on re-rendering), `self.pages` may have a better usable value than `self.ui.document:getPageCount()`.
(The joys of ReaderFooter.... rule of survival : touch as little as possible :)